packages/@ember/-internals/glimmer/lib/component-managers/curly.ts

```diff
@@ -331,6 +331,10 @@ export default class CurlyComponentManager
         instance: component,
         template: state.template,
       });
+
+      registerDestructor(bucket, () => {
+        environment.extra.debugRenderTree.willDestroy(bucket);
+      });
```

Comment:
Shouldn't this be wrapped (the removed code was)? 

```suggestion
      if (ENV._DEBUG_RENDER_TREE) {
        registerDestructor(bucket, () => {
          environment.extra.debugRenderTree.willDestroy(bucket);
        });
      }
```

Reply:
It is, if you expand the diff a bit, I combined it with the pre-existing `ENV._DEBUG_RENDER_TREE` setup code that existed in the `create` hook.